### PR TITLE
New contact method

### PR DESCRIPTION
**Pull request by cristipaval**
### Details
cc @Beamanator 

### Fixed Issues
$ https://github.com/Expensify/App/issues/15318

### Tests and QA Steps
1. Sign in and go to Profile -> Contact Method
2. Add new contact method which isn't claimed by another user
3. Verify that you are redirected to Contact Methods and the newly added contact method is in the list
4. Validate it and verify that the green indicator goes away
5. Add a new contact method which is added already to another account
6. Verify that you are redirected to the contact methods list and the newly added one has a red indicator with an error
7. Verify that you can successfully remove the contact methods added in the steps above

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Go to Contact Methods on deviceA
2. Sign in with the same account and go to contact methods on deviceB
3. Turn off the internet on deviceA
4. Add a new contact method
5. Verify that you are redirected back to Contact Methods and the newly added contact method is in a _pending_ action
6. Turn on the internet on deviceA
7. Verify that the newly added contact method is not greyed out anymore and you can click and validate it
8. Verify on deviceB that the new contact method is listed in Contact Methods page

https://user-images.githubusercontent.com/18078393/228856494-1ea79db9-b90a-44f6-82d1-a63c731146a4.mov

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/18078393/228631161-b438e6ec-98ac-4012-9362-8c8cae97fb48.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/18078393/228631209-03d59c0a-9c84-4533-886a-5fd5ee8984a3.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/18078393/228631255-41f1327c-7d5b-48a9-b3f9-9e54abbf8f95.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/18078393/228631293-24561f57-2b5b-43cc-b0aa-ba50d0ecfc7c.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/18078393/228631332-ddb3dcdf-14ec-4e9b-abbe-a791c437a53b.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/18078393/228634940-f4d66f70-ea88-4180-aa1e-d0763f6bba25.mov

</details>

### CODE REVIEW

**Comment by cristipaval** [comment]
Not overdue, waiting for a [reverted PR](https://github.com/Expensify/App/pull/15204) to be merged back into main

**Comment by marcaaron** [comment]
Can I do anything to help unblock you here @cristipaval ?

**Comment by cristipaval** [comment]
This is held on [this PR](https://github.com/Expensify/App/pull/15751)

**Comment by Beamanator** [comment]
Yo you're planning to delete the AddSecondaryLoginPage in this PR right?

**Comment by cristipaval** [comment]
> Yo you're planning to delete the AddSecondaryLoginPage in this PR right?

Yes.

**Review by marcochavezf — DISMISSED** [review]

File: src/languages/es.js
Comment (marcaaron) [reply src/languages/es.js:373]: +1 I am gonna guess "Concierge" is part of our brand identity which sort of transcends language, but maybe @Expensify/marketing has the final word on that...

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.js
Comment (marcaaron) [inline src/pages/settings/Profile/Contacts/NewContactMethodPage.js:100]: Is this possible? What are the cases where the ref would not yet be available, but the transition ended? :confused-dog:

**Review by marcaaron — CHANGES_REQUESTED** [review]
This looks great! I have some small comments and a few requests. 🙇

**Comment by rosegrech** [comment]
not sure why I am in this GH so gonna unsubscribe., if I am needed here ping me here. thanks

File: src/ROUTES.js
Comment (Beamanator) [inline src/ROUTES.js:55]: To keep this consistent with above routes

```suggestion
    SETTINGS_NEW_CONTACT_METHOD: `${SETTINGS_CONTACT_METHODS}/new`,
```

File: src/libs/actions/User.js
Comment (Beamanator) [reply src/libs/actions/User.js:269]: Agreed - buttttt also where are we basically deleting `isLoading: true`? I don't even see us using this loading indicator anywhere in ContactMethodsPage, NewContactMethodPage, or ContactMethodDetailsPage so maybe we can remove this?

File: src/libs/actions/User.js
Comment (Beamanator) [inline src/libs/actions/User.js:308]: Similarly, I think we can remove this line since it's already set in the optimisticData

**Review by Beamanator — CHANGES_REQUESTED** [review]
Overall looking groovy! I'm pumped to get rid of so much old login page code :D

Similar to marc, I have a few small comments but nothing big 👍

File: src/libs/actions/User.js
Comment (marcaaron) [reply src/libs/actions/User.js:269]: Oh good point - yes let's 🔪 everything we can

File: src/languages/es.js
Comment (joaniew) [reply src/languages/es.js:373]: i think we should keep it as Concierge if we're treating it like a proper noun (e.g. a name or a brand name) but I put it in here to discuss in case anyone feels otherwise https://expensify.slack.com/archives/C21FRDWCV/p1680657688103779

Comment (cristipaval) [reply src/languages/es.js]: fixed by reusing another existing key `common.conciergeHelp`

Comment (cristipaval) [reply src/languages/es.js]: Thank you all for your thoughts!

File: src/libs/actions/User.js
Comment (cristipaval) [reply src/libs/actions/User.js:219]: 💪

File: src/libs/actions/User.js
Comment (cristipaval) [reply src/libs/actions/User.js:269]: Oops! Bad copy-pasta 👎 
So lucky to have you guys!

File: src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js
Comment (cristipaval) [reply src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js]: fixed.

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.js
Comment (cristipaval) [reply src/pages/settings/Profile/Contacts/NewContactMethodPage.js:100]: I don't know when exactly this is possible but I see that we always check if the input ref is null when we auto-focus at the end of the screen transition.

File: src/pages/settings/Profile/Contacts/ContactMethodsPage.js
Comment (cristipaval) [reply src/pages/settings/Profile/Contacts/ContactMethodsPage.js:94]: 👌

File: src/pages/settings/Profile/Contacts/LoginField.js
Comment (cristipaval) [reply src/pages/settings/Profile/Contacts/LoginField.js:1]: 👋

**Comment by cristipaval** [comment]
Ready for a new round of reviews, @marcaaron @Beamanator @marcochavezf.

Adding @shawnborton as well for reviewing the UI

File: src/libs/actions/User.js
Comment (marcaaron) [reply src/libs/actions/User.js:191]: Good instinct and I think it sounds like a valid Follow Up thing to me.

File: src/libs/actions/User.js
Comment (marcaaron) [inline src/libs/actions/User.js:269]: `Block must not be padded by blank lines`

Sorry I've always wanted to feel what it was like to be a linter... if even just for a fleeting moment.

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.js
Comment (marcaaron) [reply src/pages/settings/Profile/Contacts/NewContactMethodPage.js:100]: Ah yes, well, sometimes the code is immortal. I've seen these before. Things in the code that nobody knows why we do them so they have life eternal. Nobody is brave enough to challenge them.

**Review by marcaaron — DISMISSED** [review]
This is looking good! I haven't tested. Will @mananjadhav do the testing?

**Comment by shawnborton** [comment]
Where is this particular style coming from?
image

Is there a mockup you are working off of? the arrow feels a bit too far to the left, and I wonder if we even need an arrow there or need to make the button green since this isn't really a GBR thing.

**Comment by cristipaval** [comment]
> Where is this particular style coming from? image
> 
> Is there a mockup you are working off of? the arrow feels a bit too far to the left, and I wonder if we even need an arrow there or need to make the button green since this isn't really a GBR thing.

hey, @shawnborton, I followed [these](https://docs.google.com/document/d/17akqrbC-w8FqEoDLvcLmrX6iw4uaVrT_xlwAl-IhEzE/edit#bookmark=id.70gptucpwl7u) mockups which are very old. We don't have an updated Figma file. I basically reused the button that we have in the App. The other place where the `+` left icon is used in the green button is on the page where the user chooses the bank account where to transfer the balance.

File: src/libs/actions/User.js
Comment (cristipaval) [reply src/libs/actions/User.js]: Thanks!

File: src/libs/actions/User.js
Comment (Beamanator) [reply src/libs/actions/User.js:191]: 👍 Added as a polish task!

**Review by Beamanator — DISMISSED** [review]
This is honestly looking great to me! I think we're waiting for @mananjadhav to do all the testing on all platforms though, right?

**Comment by marcaaron** [comment]
I have a few NewDot PRs to test today anyway so I'm going to add this to my list so we can push it through.

**Comment by marcaaron** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by marcaaron** [comment]
Testing well on web. I had just a couple of polish points so far:

1. The design of the button looks a bit off. I think maybe the + is too close to the left edge. Or maybe the text is too far from
 the +. Something is weird about it cc @shawnborton 



2. When removing a contact method that has an error I think this alert is a bit too much:

**Comment by marcaaron** [comment]
>add the user as your Copilot instead

This copy is also slightly confusing since there's no way to add a copilot in NewDot (AFAIK), but maybe that's OK. Just calling stuff out as I see it.

**Comment by cristipaval** [comment]
> > add the user as your Copilot instead
> 
> This copy is also slightly confusing since there's no way to add a copilot in NewDot (AFAIK), but maybe that's OK. Just calling stuff out as I see it.

@marcaaron where did you find this copy?

**Comment by marcaaron** [comment]
Shown here:

**Comment by cristipaval** [comment]
@marcaaron this is ready for a new review.

**Review by marcaaron — APPROVED** [review]
Looks great! Nice work. Gonna merge this so we can keep charging!

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.js
Comment (Santhosh-Sellavel) [inline src/pages/settings/Profile/Contacts/NewContactMethodPage.js:91]: We didn't trim the login value, which caused this bug [here](https://github.com/Expensify/App/issues/17770#issuecomment-1535687452)

**Comment by s77rt** [comment]
This PR caused a bug https://github.com/Expensify/App/issues/18207. The [submitted login](https://github.com/Expensify/App/blob/539fa785a30029ed74a6a982fe5f1377f01d4107/src/pages/settings/Profile/Contacts/NewContactMethodPage.js#L91) should have the sms domain appended in case of a phone number (we could have achieved using `addSMSDomainIfPhoneNumber`).

File: src/pages/settings/Profile/Contacts/NewContactMethodPage.js
Comment (eVoloshchak) [inline src/pages/settings/Profile/Contacts/NewContactMethodPage.js:115]: 👋 This has caused a small regression in https://github.com/Expensify/App/issues/18003
Since this is an email/phone TextInput, it should've had the `email` keyboard type

File: src/libs/actions/User.js
Comment (Beamanator) [inline src/libs/actions/User.js:313]: Not lowercasing the `contactMethod` caused issues when people started testing mixed-case contact methods - https://github.com/Expensify/App/issues/19080

**Pull request merged**
